static/src/javascripts/projects/common/modules/commercial/dfp/dfp-api.js

```diff
@@ -471,34 +485,43 @@ define([
         if (adverts.length === 0) {
             disableLazyLoad();
         } else {
-            var scrollTop = window.pageYOffset,
-                viewportHeight = bonzo.viewport().height,
-                scrollBottom = scrollTop + viewportHeight,
-                depth = 0.5;
-
-            chain(adverts).and(keys).and(filter, function (adSlotId) {
-                return !adverts[adSlotId].isLoading &&
-                       !adverts[adSlotId].isRendered &&
-                       // if the position of the ad is above the viewport - offset (half screen size)
-                    scrollBottom > document.getElementById(adSlotId).getBoundingClientRect().top + scrollTop - viewportHeight * depth;
-            }).and(forEach, function (slot) {
-                loadSlot(slot);
+            var scrollTop = window.pageYOffset;
+            var viewportHeight = bonzo.viewport().height;
+            var scrollBottom = scrollTop + viewportHeight;
+            var depth = 0.5;
+
+            var advertsToLoad = values(adverts).filter(function (advert) {
+                return !advert.isRendered
+                    && !advert.isLoading
+                    // if the position of the ad is above the viewport - offset (half screen size)
+                    && (scrollBottom > document.getElementById(advert.adSlotId).getBoundingClientRect().top + scrollTop - viewportHeight * depth);
+            });
+
+            advertsToLoad.forEach(function (advert) {
+                loadSlot(advert);
             });
         }
     }
 
     function instantLoad() {
-        chain(adverts).and(keys).and(forEach, function (adSlotId) {
-            if (contains(['dfp-ad--pageskin-inread', 'dfp-ad--merchandising-high', 'dfp-ad--im'], adSlotId)) {
-                loadSlot(adSlotId);
+        values(adverts).forEach(function (advert) {
```

Comment:
It's great to finally see array functions used instead of the lodash counterparts. Why not go all the way and use: `Object.keys(adverts).forEach(...)`? I feel like `values()` is one we could get rid of very easily.